lms/djangoapps/lti_provider/views.py

```diff
@@ -64,8 +66,14 @@ def lti_launch(request, course_id, usage_id):
     # Store the course, and usage ID in the session to prevent privilege
     # escalation if a staff member in one course tries to access material in
     # another.
-    params['course_id'] = course_id
-    params['usage_id'] = usage_id
+    try:
+        course_key, usage_key = parse_course_and_usage_keys(course_id, usage_id)
+    except InvalidKeyError:
+        raise Http404(
+            'Invalid course key {} or usage key {}'.format(course_id, usage_id)
```

Comment:
I actually don't remember what we show users when we 404 these days, but it's likely safe to just use it. If you want to read up on i18n in edX, please see:

http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/internationalization/i18n.html#python-source-code

I've linked to the most relevant sub-section, but the entire thing is good to read when you have time.